### ISSUE TITLE
Error 1006 - Access denied; Images are blocked by Cloudflare (automatic rate limit)

### ISSUE THREAD

**Issue by ndavoust** [andrieslouw/imagesweserv#168]
Hi, 

all our images are blocked (for instance https://images.weserv.nl/?url=gensdeconfiance.s3.amazonaws.com%2Fimages%2Fmembers%2Fthumb%2F2016_01_21_56a118ccdc0bb.jpg&t=squaredown&w=320&h=0)

it seems that it is just us, for our ip.
![image](https://user-images.githubusercontent.com/404921/49581329-d53cd100-f951-11e8-91e7-509400be080f.png)

is it possible to whitelist us?

thanks a lot!

nicolas

**Comment by andrieslouw** [comment]
Please see https://github.com/andrieslouw/imagesweserv/issues/83, https://github.com/weserv/images/issues/124 

If needed, we could make an exception for your IP-address, but I would like to know a rough estimation of usage, to check if our servers will be able to cope.

**Comment by kleisauke** [comment]
We've refined our rate limiter in API version 5. It's now easy to check show your current rate limit quota:
https://wsrv.nl/news/2019/09/01/introducing-api-5.html#improved-rate-limiter

**Comment by andrieslouw** [comment]
Closing due to inactivity. Let us know if there is anything we can do to assist.

**Comment by planetahuevo** [comment]
I am reading this old thread about rate limiting and the link is no longer valid. 
Can someone point me to the right docs? Or maybe this is not relevant anymore?

**Comment by kleisauke** [comment]
I've updated the link above, thanks for the heads up.

**Comment by planetahuevo** [comment]
Fantastic. Is there any guide on how to change those limits for a custom install?
Do not answer here. Probably editing the code... I will check and ask on another issue if needed. 
(Also, those limits are huge, so no need to edit them for sure.)

Thanks!

**Comment by kleisauke** [comment]
@planetahuevo If you are referring to the Dockerfile, the rate limiter is deliberately disabled in that by default, for example:
https://github.com/weserv/images/blob/03684dc35801188ef12feed5219daf9da4b4c919/ngx_conf/imagesweserv.conf#L145-L147

This was done to avoid using a process manager like `supervisord`, which is a bit of a heavy-weight approach, see comment https://github.com/weserv/images/issues/180#issue-438049613.

**Comment by planetahuevo** [comment]
Oh, interesting. 
I can see that it has been disabled, but is there any docs on how to enable it with the supervisord? 
I suppose we need to remove the comment on that file, but also activate supervisord as well. 
I think is worth to have the rate limiting even if that means having a little bit of heavy weight. If you are using it, there is a reason. :)
